dns/src/config/mod.rs

```diff
@@ -0,0 +1,77 @@
+mod file;
+mod structs;
+
+use crate::http::Domain;
+use colored::Colorize;
+use macros_rs::fmt::{crashln, string};
+use mongodb::{options::ClientOptions, Client};
+use std::fs::write;
+use structs::{Mongo, Server, Settings};
+
+pub use structs::Config;
+
+impl Config {
+    pub fn new() -> Self {
+        let default_offensive_words = vec!["nigg", "sex", "porn", "igg"];
+        let default_tld_list = vec![
+            "mf", "btw", "fr", "yap", "dev", "scam", "zip", "root", "web", "rizz", "habibi", "sigma", "now", "it", "soy", "lol", "uwu", "ohio", "fur",
+        ];
+
+        Config {
+            config_path: "config.toml".into(),
+            server: Server {
+                address: "127.0.0.1".into(),
+                port: 8080,
+                key_db: "storage".into(),
+                mongo: Mongo {
+                    connection: "".into(),
+                    app_name: "DomainApp".into(),
+                    db_name: "mydb".into(),
+                },
+            },
+            settings: Settings {
+                tld_list: default_tld_list.iter().map(|s| s.to_string()).collect(),
+                offensive_words: default_offensive_words.iter().map(|s| s.to_string()).collect(),
+            },
+        }
+    }
+
+    pub fn read(&self) -> Self { file::read(&self.config_path) }
+    pub fn get_address(&self) -> String { format!("{}:{}", self.server.address.clone(), self.server.port) }
+    pub fn tld_list(&self) -> Vec<&str> { self.settings.tld_list.iter().map(AsRef::as_ref).collect::<Vec<&str>>() }
+    pub fn offen_words(&self) -> Vec<&str> { self.settings.offensive_words.iter().map(AsRef::as_ref).collect::<Vec<&str>>() }
+
+    pub fn set_path(&mut self, config_path: &String) -> &mut Self {
+        self.config_path = config_path.clone();
+        return self;
+    }
+
+    pub fn write(&self) -> &Self {
+        let contents = match toml::to_string(self) {
+            Ok(contents) => contents,
+            Err(err) => crashln!("Cannot parse config.\n{}", string!(err).white()),
+        };
+
+        if let Err(err) = write(&self.config_path, contents) {
+            crashln!("Error writing config to {}.\n{}", self.config_path, string!(err).white())
+        }
+
+        log::info!("Created config: {}", &self.config_path,);
+
+        return self;
+    }
+
+    pub async fn connect_to_mongo(&self, mongo: &crate::DB) {
+        let mut client_options = ClientOptions::parse(&self.server.mongo.connection).await.unwrap();
```

Comment:
Wouldn't this throw an error if the client fails to connect? Might be worth using `crashln!` because we *need* the DB to be here to run?

Reply:
Fixing, forgot to add this

Reply:
I'm sure you know this, @theMackabu, but despite my name, `.unwrap()` should rarely -- if ever -- be used in a production app. We definitely are going to want graceful error handling here to avoid `panic`

Reply:
its the one part I missed and forgot to use crashln!

Reply:
> Btw if im not wrong this might return Ok even if there's no connection to the db and will error on running a command like find_one

Read the thread. They are fixing it.